src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
```

Comment:
PullProofAI

The conditional logic in the `APIResponse.create` function can be refactored for better clarity and maintainability. Instead of checking for `error` and `data` separately, you can simplify the logic by directly assigning `data` or `error` based on conditions. This will make the code easier to understand and maintain.

```typescript
// Refactored APIResponse.create function
export const APIResponse = {
  create: (
    isSuccess: number,
    statusCode: number,
    message?: string,
    data?: any,
    error?: any,
  ): ControllerResponse => {
    const response: ControllerResponse = { isSuccess, statusCode, message };

    if (error) {
      response.error = error;
    } else {
      response.data = data || {};
    }

    return response;
  },
};
```
By simplifying the conditional logic, the code becomes more concise and easier to follow, enhancing readability and maintainability.


<!-- This is an auto-generated comment by PullProofAI -->